crates/ruff_workspace/src/options.rs

```diff
@@ -2381,11 +2390,141 @@ impl PyUpgradeOptions {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum FormatOrOutputFormat {
+    Format(FormatOptions),
+    OutputFormat(SerializationFormat),
+}
+
+impl FormatOrOutputFormat {
+    pub const fn metadata() -> crate::options_base::OptionGroup {
+        FormatOptions::metadata()
+    }
+
+    pub const fn as_output_format(&self) -> Option<SerializationFormat> {
+        match self {
+            FormatOrOutputFormat::Format(_) => None,
+            FormatOrOutputFormat::OutputFormat(format) => Some(*format),
+        }
+    }
+}
+
+#[derive(
+    Debug, PartialEq, Eq, Default, Serialize, Deserialize, ConfigurationOptions, CombineOptions,
+)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct FormatOptions {
+    /// A list of file patterns to exclude from formatting.
+    ///
+    /// Exclusions are based on globs, and can be either:
+    ///
+    /// - Single-path patterns, like `.mypy_cache` (to exclude any directory
+    ///   named `.mypy_cache` in the tree), `foo.py` (to exclude any file named
+    ///   `foo.py`), or `foo_*.py` (to exclude any file matching `foo_*.py` ).
+    /// - Relative patterns, like `directory/foo.py` (to exclude that specific
+    ///   file) or `directory/*.py` (to exclude any Python files in
+    ///   `directory`). Note that these paths are relative to the project root
+    ///   (e.g., the directory containing your `pyproject.toml`).
+    ///
+    /// For more information on the glob syntax, refer to the [`globset` documentation](https://docs.rs/globset/latest/globset/#syntax).
+    ///
+    /// Note that you don't need to list files that are excluded by
+    /// [`exclude`](#exclude).
+    // TODO(micha): Document once the CLI supports the exclude option
+    // #[option(
+    //     default = r#"[]"#,
+    //     value_type = "list[str]",
+    //     example = r#"
+    //         [format]
+    //         exclude = ["generated"]
+    //     "#
+    // )]
+    pub exclude: Option<Vec<String>>,
+
+    /// Whether to enable the unstable preview style formatting.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            [format]
```

Comment:
These end up appearing under a `[tool.ruff]`, so I think these should be `[tool.ruff.format]`?